helm/charts/determined/templates/genai/genai-initialize-shared-fs-permissions.yaml

```diff
@@ -0,0 +1,53 @@
+{{- if .Values.genai }}
+{{- if .Values.genai.version }}
+{{- /* Helm Job to make sure that the shared filesystem sets up group permissions for */ -}}
+{{- /* the all members of the group defined in .Values.genai.generatedPVC.agentGroupID */ -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: genai-initialize-shared-fs-permissions-{{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: genai-{{ .Release.Name }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "0"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: genai-initialize-shared-fs-permissions-{{ .Release.Name }}
+      labels:
+        app: genai-initialize-shared-fs-permissions-{{ .Release.Name }}
+        release: {{ .Release.Name }}
+    spec:
+      serviceAccount: determined-master-{{ .Release.Name }}
+      restartPolicy: OnFailure
+      {{ $gid := (required "A valid .Values.genai.generatedPVC.agentGroupID entry required!" .Values.genai.generatedPVC.agentGroupID) }}
+      securityContext:
+        runAsUser: 0
```

Comment:
This is my main concern. I don't know how else to run this without at least this pod running as root to set up the shared_fs the first time when you run the helm install. 

Is there a way around this? Are there use cases or clusters where this wouldn't be allowed at all?

Reply:
>Are there use cases or clusters where this wouldn't be allowed at all?

yes, this would be disallowed on a number of clusters. the only alternative I can think of is to tell admins to do it manually.

Reply:
So doesn't this mean we shouldn't do this entirely? Theoretically this would mean that when they try to use `helm install` on a cluster that would restrict root access, I think it would block the whole helm deployment until that job finishes. I can confirm but I think that's the behavior of these hooks.

I was writing up documentation for sys admins to handle this themselves and I'm starting to think that it's the correct path forward if we are dealing with clusters that would not allow any form of root pod.

https://hpe-aiatscale.atlassian.net/wiki/spaces/~63bdb6698a7d2f693bf6cb21/pages/1478819920/GenAI+Kubernetes+k8s+Installation

Reply:
I think that what I could do, is make this job configurable as to whether it runs in the values

Reply:
>I think that what I could do, is make this job configurable as to whether it runs in the values
yeah that may be the most flexible option.